srv/src/main/java/my/bookshop/handlers/CatalogServiceHandler.java

```diff
@@ -34,14 +35,23 @@ public CatalogServiceHandler(PersistenceService db) {
 
 	@After(event = CdsService.EVENT_READ)
 	public void discountBooks(Stream<Books> books) {
-		books.filter(b -> b.getTitle() != null).filter(b -> {
-			Integer stock = b.getStock();
-			if (stock == null) {
-				stock = db.run(Select.from(BOOKS).byId(b.getId()).columns(c -> c.stock())).single(Books.class).getStock();
+		books.filter(b -> b.getTitle() != null).forEach(
+			b -> {
+				loadStockIfNotSet(b);
+				discountBooksWithMoreThan111Stock(b);
 			}
-			return stock > 111;
-		})
-		.forEach(b -> b.setTitle(b.getTitle() + " -- 11% discount"));
+		);
 	}
 
+	private void discountBooksWithMoreThan111Stock(Books b) {
+		if(b.getStock() > 111) {
+			b.setTitle(String.format("%s -- 11%% discount", b.getTitle()));
+		}
+	}
+
+	private void loadStockIfNotSet(Books b) {
+		if (b.getStock()== null) {
```

Comment:
```suggestion
		if (b.getStock() == null) {
```

Reply:
Obviously, I have no code formatting rules installed, yet. ;)

Reply:
We also didn't strictly define any so far for this project.. ;)

Reply:
I corrected the whitespace but now the build fails. 🤯 😄 

I tried the build on my Mac (maven 3.6 and jdk 8) and its runs without failure.